src/components/controls/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+
+import {Eye, EyeSlash} from '@gravity-ui/icons';
+
+import {ActionTooltip} from '../../ActionTooltip';
+import {Button} from '../../Button';
+import {ClipboardButton} from '../../ClipboardButton';
+import {Icon} from '../../Icon';
+import {block} from '../../utils/cn';
+import {TextInput} from '../TextInput';
+import type {TextInputProps} from '../TextInput';
+
+import {i18n} from './i18n';
+import {getActionButtonSizeAndIconSize} from './utils';
+
+import './PasswordInput.scss';
+
+const b = block('password-input');
+
+export type PasswordInputProps = Required<Pick<TextInputProps, 'onUpdate' | 'value'>> &
+    Omit<TextInputProps, 'type'> & {
+        /** Show copy button */
+        showCopyButton?: boolean;
+        /** Show reveal button */
+        showRevealButton?: boolean;
+        /** Disable the tooltip for the copy button. The tooltip will not be displayed */
+        hasCopyTooltip?: boolean;
+        /** Disable the tooltip for the reveal button. The tooltip will not be displayed */
+        hasRevealTooltip?: boolean;
+    };
+
+export const PasswordInput = (props: PasswordInputProps) => {
+    const {
+        autoComplete,
+        value,
+        showCopyButton,
+        rightContent,
+        endContent,
+        showRevealButton,
+        size = 'm',
+        hasCopyTooltip = true,
+        hasRevealTooltip = true,
+        controlProps,
+    } = props;
+
+    const [hideValue, setHideValue] = React.useState(true);
+
+    const additionalEndContent = React.useMemo(() => {
+        if (!showRevealButton && !showCopyButton) {
+            return <React.Fragment>{endContent || rightContent}</React.Fragment>;
+        }
+
+        const onClick = () => {
+            setHideValue((hideValue) => !hideValue);
+        };
+
+        const {actionButtonSize, iconSize} = getActionButtonSizeAndIconSize(size);
+
+        return (
+            <div className={b('additional-end-content')}>
```

Comment:
we can remove this wrapper, additional content wrapper in the TextInput already has the same styles. SO we can simply write 
```ts
const additionalEndContent = (
    <React.Fragment>
        {endContent || rightContent}
        {value && showCopyButton ? ... : null}
        {showRevealButton ? ... : null}
    </React.Fragment>
);

Reply:
thank you! When I made this component in https://github.com/gravity-ui/components they were necessary